### ISSUE TITLE
Pass index to slot of Vue Virtualizer

### ISSUE THREAD

**Issue by inokawa** [inokawa/virtua#498]
Any plans to pass index to slot in Vue implementation? We really need this feature in Vue!

_Originally posted by @melon in https://github.com/inokawa/virtua/issues/496#issuecomment-2290795545_

**Comment by inokawa** [comment]
It will be like vue-virtual-scroller.

**Comment by melon** [comment]
> https://github.com/inokawa/virtua/pull/496#issuecomment-2290814156

I think it could be a breaking change because scoped slots usually have only one parameter. But if we use multiple parameters in scoped slots, then the solution could be both simple and backward compatible.

We could just add the index parameter as the second parameter of slots.default(), just as follows.

https://github.com/inokawa/virtua/blob/70149236634a031ce9b50980d45a8d922859c032/src/vue/Virtualizer.tsx#L249

```
 const e = slots.default(props.data![i]!, i)[0]! as VNode; 
```

And then use the scope like the following way:

```
<VList
  :data="data"
  #default="item, index"
>
{{ item }} {{ index }}
</VList>
```

Well, although there're no examples or guidelines on multiple parameters in scoped slots on the vue's official documentation, multiple parameters just works as magic. There has been some discussions arguing whether scoped slots should have only one parameter or not.

> https://github.com/vuejs/eslint-plugin-vue/issues/722

And it seems that the vue team has already officially added the rules for multiple parameters in scoped slots.

**Comment by inokawa** [comment]
I didn't know that. That looks good.

**Comment by inokawa** [comment]
https://github.com/vuejs/vue/issues/9468#issuecomment-462210146

**Comment by inokawa** [comment]
Released in `0.34.0`.
